tools/create_collection/create_collection.xml

```diff
@@ -0,0 +1,87 @@
+<tool id="create_collection" name="Create_dataset_collection" version="0.3.0">
+  <description>of type list</description>
+  <stdio>
+      <exit_code range="1:" level="fatal" description="Tool exception" />
+  </stdio>
+  <command detect_errors="exit_code"><![CDATA[
+       mkdir outputs; cd outputs;
+       #set names={}#
+       #for $input in $input_list#
+         #set $file_name =$input.name#
+           #if $file_name in $names#
+             #silent $names[$file_name] = $names[$file_name] + 1#
+             #set $file_name = ''.join([$file_name,'(',str($names[$file_name]-1),')'])#
+           #else#
+             #silent $names[$file_name] = 0#
+           #end if#
+         echo "$file_name" >> $log;
+         ln  -s  $input "$file_name.$input.ext.$input.dbkey";
```

Comment:
The symlinks will break on galaxy instances that use more than one object stores which might be on different disks.

There will however be a couple of new collection tools in the upcoming 18.09 release (https://github.com/galaxyproject/galaxy/pull/6312/files) that will allow you to create collections form datasets and datasets from collections (in workflows or otherwise) without increasing disk consumption. 

If you want to beta test them feeedback is always very welcome!